module/Activity/src/Activity/Form/Activity.php

```diff
@@ -16,13 +21,22 @@ class Activity extends Form implements InputFilterProviderInterface
     protected $inputFilter;
     protected $organs;
 
-    public function __construct()
+    public function __construct(Member $user, Translator $translator)
     {
         parent::__construct('activity');
         $this->setAttribute('method', 'post');
         $this->setHydrator(new ClassMethodsHydrator(false))
             ->setObject(new \Activity\Model\Activity());
 
+        // all the organs that the user belongs to in organId => name pairs
+        $organs = [0 => $translator->translate('No organ')];
+
+        // Find user that wants to create an activity
+        $user->getCurrentOrganInstallations()->forAll(function ($index, OrganMember $organMember) use (&$organs) {
+            $organ = $organMember->getOrgan();
+            $organs[$organ->getId()] = $organ->getName();
+        });
+
```

Comment:
This kinds of stuff would not be needed if we query the database directly for the data we need.